python/lsst/afw/table/schema/schema.cc

```diff
@@ -82,6 +82,48 @@ void declareFieldBaseSpecializations(PyFieldBase<std::string> &cls) {
     cls.def("getSize", &FieldBase<std::string>::getSize);
 }
 
+// Specializations for Field
+
+template <typename T>
+void declareFieldSpecializations(PyField<T> &cls) {
+    cls.def(py::pickle(
+            [](Field<T> const &self) {
+                /* Return a tuple that fully encodes the state of the object */
+                return py::make_tuple(self.getName(), self.getDoc(), self.getUnits());
+            },
+            [](py::tuple t) {
+                if (t.size() != 3) throw std::runtime_error("Invalid number of parameters when unpickling!");
+                return Field<T>(t[0].cast<std::string>(), t[1].cast<std::string>(), t[2].cast<std::string>());
+            }));
+}
+
+template <typename U>
+void declareFieldSpecializations(PyField<Array<U>> &cls) {
+    cls.def(py::pickle(
+            [](Field<Array<U>> const &self) {
```

Comment:
Consider `using Class = Field<Array<U>>` to reduce code duplication; see the [style guide](https://developer.lsst.io/pybind11/style.html#names-of-generic-class-types-should-be-called-class).

Reply:
I think there are few enough uses (3 total) that that definition doesn't help, and it gives another layer of indirection for someone looking at the code to have to follow.